Src/WitsmlExplorer.Frontend/services/credentialsService.ts

```diff
@@ -78,76 +83,69 @@ class CredentialsService {
   }
 
-  public clearPasswords() {
-    this.credentials = this.credentials.map((creds) => {
-      if (this.hasValidCookieForServer(creds.server.url)) {
-        return creds;
-      }
-      return { server: creds.server };
-    });
-    if (!this.hasValidCookieForServer(this.server.url)) {
-      this._onCredentialStateChanged.dispatch({ server: this.server, hasPassword: this.isAuthorizedForServer(this.server) });
-    }
-  }
-
   public isAuthorizedForServer(server: Server): boolean {
     if (msalEnabled && server?.securityscheme == SecurityScheme.OAuth2 && getUserAppRoles().some((x) => server.roles.includes(x))) {
       return true;
     }
-    return this.credentials.find((c) => c.server.id === server?.id)?.password !== undefined;
-  }
-
-  public hasPasswordForUrl(serverUrl: string) {
-    return this.credentials.find((c) => c.server.url === serverUrl)?.password !== undefined;
+    return this.hasValidCookieForServer(server?.url);
   }
 
   public hasValidCookieForServer(serverUrl: string): boolean {
     //use local storage to check whether the cookie is valid, because the cookie is httpOnly
-    const cookieTimestamp = localStorage.getItem(serverUrl);
-    if (!cookieTimestamp) {
+    const cookieInfo = localStorage.getItem(serverUrl);
+    if (!cookieInfo) {
       return false;
+    } else {
+      const cInfo: CookieInfo = JSON.parse(cookieInfo);
+      return new Date().getTime() < new Date(cInfo.expiry).getTime();
     }
-    return new Date().getTime() < new Date(cookieTimestamp).getTime();
-  }
-
-  public keepLoggedInToServer(serverUrl: string): boolean {
-    return !!localStorage.getItem(serverUrl);
   }
 
-  public async verifyCredentials(credentials: BasicServerCredentials, abortSignal?: AbortSignal): Promise<any> {
-    const response = await ApiClient.get(`/api/credentials/authorize`, abortSignal, [credentials]);
-    if (response.ok) {
-      return response.json();
-    } else {
-      const { message }: ErrorDetails = await response.json();
-      CredentialsService.throwError(response.status, message);
+  private isJsonString(str: string) {
+    try {
+      JSON.parse(str);
+    } catch (e) {
+      return false;
     }
+    return true;
   }
 
-  public async verifyCredentialsWithCookie(credentials: BasicServerCredentials, abortSignal?: AbortSignal): Promise<BasicServerCredentials> {
-    const response = await ApiClient.get(`/api/credentials/authorizewithcookie`, abortSignal, [credentials], true);
-    if (response.ok) {
-      const cookie = await response.json();
-      const decoded = Buffer.from(cookie, "base64").toString();
-      const creds = decoded.split(":");
-      return {
-        server: credentials.server,
-        username: creds[0],
-        password: creds[1]
-      };
-    } else {
-      const { message }: ErrorDetails = await response.json();
-      CredentialsService.throwError(response.status, message);
+  private allLocalStorage() {
```

Comment:
Why not just loop over credentials like in deauthorize()?

Reply:
that might be a better approach, I will check that out

Reply:
Agreed in meeting 27.10 to keep as is for now to be re-assessed in WE-614